docs/plugins/events.md

```diff
@@ -1,45 +1,138 @@
 # Events
 
-Generic Events emitter / listener interface which can be used for _any_ type of event. For Lifecycle events please see [Lifeycle](/plugins/lifecycle.md)
+_work in progres_
+
+Events plugin is used to listen to events emitted from:
+
+- various SDK plugins (i.e. VideoPlayer plugin, Router plugin)
+- the transport layer (for realtime changes in the Platform or Lifecycle)
+- custom App events broadcasted from the App code
 
 ## Usage
 
-The SDK comes with a generic Eventing mechanism which can be, optionally, used within the application. To start:
+In order to use the Events plugin, import it from the Lightning SDK.
 
 ```js
 import { Events } from '@lightningjs/sdk'
 ```
 
 ## Available methods
 
-### Listening for events
+### Listen
+
+Listen to events and execute a callback when they occur.
+
+Examples:
+
+1) Listen to a specific event on a specific Plugin
+
+```js
+Events.listen('Lifecycle', 'pause', () => {
+  console.log('The App is going in a pause state')
+})
+
+Events.listen('Platform', 'device.hdcp', val => {
+  console.log('The hdcp value changed on the platform!', val)
+})
+```
+
+2) Listen to all events of a specific Plugin
+
+```js
+Events.listen('Lifecycle', (val, plugin, event) => {
```

Comment:
`plugin` in the callback signature is redundant here, isn't it?

Reply:
yup .. that was bugging me a bit as well :)

ideally I'd only pass non-redundant arguments, but not sure yet if I can pull that off without too much extra code. I've been considering simply changing the order to `(val, event, plugin)`, which might be just what we need

Reply:
it's not redundant if a developer creates a function that's going to handle multiple events from different plugins, and just pass a reference to that function into `listen`. You'd need a way to fully distinguish the event, which is `plugin + event`.

maybe for the callback, we need a single object, with fields, e.g.:

```javascript
{
    value: 'foo',
    module: 'Platform',
    event: 'device.hdcp'
}
```

Reply:
I'd avoid returning objects in the absence of type declarations. As a user, it's easier for me to grok the callback signature. Plus, I _think_ you'll get better IDE behavior out of the box with a sane callback signature as opposed to an opaque object without type decs.

Reply:
I agree completely with you @cletustboone ..

I've refactored this part in this commit: https://github.com/rdkcentral/Lightning-SDK/commit/020a04a23768d9eaddabc92121929117ccd03aaf

I think this API looks nicer and makes more sense from an App dev perspective.

Reply:
I do like the way this looks w/ the listener functions declared right inside of the `listen()` call, but what if someone wants to reuse an existing function, e.g.:

```javascript
function myCustomListener(plugin, event, value) {
  if (plugin === 'Platform') {
    if (event === 'device.hdcp') {
      ...
    }
    else if (event === 'accessibility.closedCaptionsToggled') {
      ...
    }
  }
}

Events.listen('Platform', 'device.hdcp', myCustomListener)
Events.listen('Platform', 'accessibility.closedCaptionsToggled', myCustomListener)
```

in that case, you need all three variables passed to each version of `listen` to get it to work.

Listening to all events seems like it might not be the most efficient way around this.